go/dhcp/main.go

```diff
@@ -502,47 +507,58 @@ func (h *Interface) ServeDHCP(ctx context.Context, p dhcp.Packet, msgType dhcp.M
 
 			var Reply bool
 			var Index int
+			var Static bool
 
+			Static = false
 			// Valid IP
 			if len(reqIP) == 4 && !reqIP.Equal(net.IPv4zero) {
 				// Requested IP is in the pool ?
 				if leaseNum := dhcp.IPRange(handler.start, reqIP) - 1; leaseNum >= 0 && leaseNum < handler.leaseRange {
-					// Requested IP is in the cache ?
-					if index, found := handler.hwcache.Get(p.CHAddr().String()); found {
-						// Requested IP is equal to what we have in the cache ?
-
-						if dhcp.IPAdd(handler.start, index.(int)).Equal(reqIP) {
-							id, _ := GlobalTransactionLock.Lock()
-							if _, found = RequestGlobalTransactionCache.Get(cacheKey); found {
-								log.LoggerWContext(ctx).Debug("Not answering to REQUEST. Already processed")
-								Reply = false
-								GlobalTransactionLock.Unlock(id)
-								return answer
+					// Static assigned ip ?
+					for macaddr, position := range handler.ipAssigned {
```

Comment:
Same logic as above, and while we're there, it would be good to have a method on the handler to find if a MAC has an IP assigned to it